### PR TITLE
Update test fixtures for better usability

### PR DESCRIPTION
**Pull request by armsnyder**
Basically covers @RomanNess's suggestions on #530 

The docker-compose.yml is a bit more verbose than the script we had previously, but the benefit is that it is a standard spec that new contributors are more likely to already be familiar with.

See the [updated README](https://github.com/gitlabhq/terraform-provider-gitlab/blob/docker-compose/README.md#running-tests) for a good overview of the testing use cases that are better optimized now.

tl;dr `make testacc-up testacc`

- Replaced the start-gitlab.sh script with a docker-compose.yml file
- Optimized the testing section of the README for common use cases
- Made it so `make testacc` works by default, without needing to set `GITLAB_TOKEN` or `GITLAB_BASE_URL`. (These variables are still configurable.)
- Removed references to the "license" directory, which does not work on MacOS due to the conflict with the LICENSE file. The license is now expected to be in the project root and named Gitlab-license.txt.

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thanks for this! Hopefully I will be able to review that soon!

File: README.md [hunk 1]
```diff
@@ -17,61 +17,101 @@ Requirements
...
-$ make testacc
+$ make testacc GITLAB_TOKEN=example123 GITLAB_BASE_URL=https://example.com/api/v4
 ```
 
-### Gitlab Community Edition and Gitlab Entreprise Edition
+`GITLAB_TOKEN` must be a valid token for an account with admin privileges.
+
+#### Testing Tips
+
+* **Gitlab Community Edition and Gitlab Entreprise Edition:**
```
Comment (zachkknowbe4) [inline README.md:75]: There's a typo for Enterprise,
```
* **Gitlab Community Edition and Gitlab Enterprise Edition:**
```

Comment (armsnyder) [reply README.md]: updated

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**